### DISCUSSION TITLE
Using expressionTemplates with serilog-sinks-http

### DISCUSSION THREAD

**Discussion by espeng2** [FantasticFiasco/serilog-sinks-http#296] [Ideas]
Hi,

I was wondering if it would be possible to specify a formatter in the configuration of the http sink.
something like this, which is possible with the console and file sink:

  "formatter": {
            "type": "Serilog.Templates.ExpressionTemplate, Serilog.Expressions",
            "template": "{ {timestamp: @t, level: @l, messageTemplate: @mt, message: @m, exception: @x, properties: if IsDefined(@p[?]) then @p else undefined()} }\n"
          }

best regards,
Espen

**Comment by FantasticFiasco** [comment]
Hi @espeng2!

It looks nice! I'm guessing that this would be an alternative to the current state where you will have to implement your own [event formatter](https://github.com/FantasticFiasco/serilog-sinks-http/wiki/Event-formatters), or do you see as something completely new, something that currently isn't supported by the sink?

**Reply by espeng2** [reply]
Hi, thanks for the reply,

Maybe it is already supported, since ExpressionTemplates implement ITextFormatter. Sorry, I didn't know you could write your own event formatter like that. I will try that out and see how that compares.
 thanks!

**Reply by FantasticFiasco** [reply]
If you find that ExpressionTemplates works with this sink, please provide an example here. I'm interested in how the two could be used together.
